earthfile2llb/converter.go

```diff
@@ -930,20 +965,27 @@ func (c *Converter) SaveImage(ctx context.Context, imageNames []string, pushImag
 					NoManifestList:      noManifestList,
 				})
 		} else {
-			pcState := c.persistCache(c.mts.Final.MainState)
-			c.mts.Final.SaveImages = append(c.mts.Final.SaveImages,
-				states.SaveImage{
-					State:               pcState,
-					Image:               c.mts.Final.MainImage.Clone(),
-					DockerTag:           imageName,
-					Push:                pushImages,
-					InsecurePush:        insecurePush,
-					CacheHint:           cacheHint,
-					HasPushDependencies: false,
-					ForceSave:           c.opt.ForceSaveImage,
-					CheckDuplicate:      c.ftrs.CheckDuplicateImages,
-					NoManifestList:      noManifestList,
-				})
+			si := states.SaveImage{
+				State:               c.persistCache(c.mts.Final.MainState),
+				Image:               c.mts.Final.MainImage.Clone(),
+				DockerTag:           imageName,
+				Push:                pushImages,
+				InsecurePush:        insecurePush,
+				CacheHint:           cacheHint,
+				HasPushDependencies: false,
+				ForceSave:           c.opt.ForceSaveImage,
+				CheckDuplicate:      c.ftrs.CheckDuplicateImages,
+				NoManifestList:      noManifestList,
+				Platform:            c.platr.Materialize(c.platr.Current()),
+			}
+
+			if c.ftrs.WaitBlock {
+				shouldPush := pushImages && !c.mts.Final.Target.IsRemote() && si.DockerTag != "" && c.opt.DoSaves
```

Comment:
What's the reasoning for `!c.mts.Final.Target.IsRemote()`?

Reply:
it came from https://github.com/earthly/earthly/blob/main/builder/builder.go#L242

I'm honestly not sure why we would want this, I'm happy to remove it.

Reply:
I think this was needed to support the old 0.5 pushing logic, which did things differently when the target was remote. I believe we can remove it in the new code.

Now that I look at the original logic that you linked, it doesn't look right in that case either... It should only depend on being remote or not in 0.5. 🤔